core/src/main/java/org/mule/internal/connection/CachedConnectionProviderWrapper.java

```diff
@@ -39,5 +51,18 @@ public ConnectionValidationResult validate(Connection connection)
         }
         return getDelegate().validate(connection);
     }
+
+    /**
+     * @return a {@link AbstractPolicyTemplate} with the configured values in the Mule Application. If not configured,
+     * a default {@link AbstractPolicyTemplate} will be returned created by the {@link ConnectionProviderWrapper}
+     */
+    public AbstractPolicyTemplate getRetryPolicyTemplate()
+    {
+        if (retryPolicyTemplate == null)
```

Comment:
isn't it better to do this o the constructor and have the template being final? Or is there a technical limitation to do that?